### ISSUE TITLE
simpletransformers.classification.ClassificationModel hangs when training

### ISSUE THREAD

**Issue by rjurney** [ThilinaRajapakse/simpletransformers#1030]
**Describe the bug**

I have the following code for a ClassificationModel to do sequence classification to detect whether a name is a company name or a person name. I am loading 1MM records, 500K of each label.

The real code loads 1 MM records, but the following code is a 10K record sample. It works, so it doesn't reproduce the issue, but a larger dataset does reproduce it every single time, even with different salts on the train_test_split.

```python
import os
import numpy as np
import pandas as pd
import s3fs
import torch
from simpletransformers.classification import ClassificationModel, ClassificationArgs
from sklearn.model_selection import train_test_split


# Die if there is no GPU
assert torch.cuda.is_available()

# Load the CSV sample
df = pd.read_csv("s3://rjurneyopen/oc/sample.csv", names=["name", "label", "label_num"])
df.head()

# Shuffle the data
df = df.sample(frac=1)

# Set the right column names
df = df[["name", "label_num"]]
df.columns = ["text", "labels"]
df.head()

# Train and evaluation datasets
train_df, test_df, y_train, y_test = train_test_split(
    df, df["labels"], test_size=0.33, random_state=31337
)

# Use 8 GPUs
model_args = ClassificationArgs(
    num_train_epochs=1,
    n_gpu=8,
    dataloader_num_workers=94,
    process_count=94,
)

# Create a ClassificationModel
model = ClassificationModel("roberta", "roberta-base", args=model_args, use_cuda=True)

# Train the model
model.train_model(train_df)

# Evaluate the model
result, model_outputs, wrong_predictions = model.eval_model(test_df)

# Make predictions with the model
predictions, raw_outputs = model.predict(
    ["Russell Jurney", "Apple", "Apple Inc.", "LegalZoom"]
)
```

I am running this code on a 96 core, 8 V-100 GPU machine with gobs of RAM. The cores get used at first but then all Python pids fall to 0% and it hangs at 38-40% in Jupyter or in iPython:

```python
model.train_model(train_df)
    ...:

INFO:simpletransformers.classification.classification_utils: Converting to features started. Cache is not used.
 40%|██████████████████████████████████████████████████████████████████████████████████████▍
```

If I hit enter the bar does not update. The time never moves. It is stuck. What do I do? :(

**To Reproduce**

Execute code above on 1MM 50/50 binary labels.

**Expected behavior**

I expect it to use actual CPU and GPU cores to perform the training of the model.

**Screenshots**

**Desktop (please complete the following information):**
 - OS - Ubuntu Linux
 - Anaconda Python v3.8.8, conda environment but dependencies installed with `pip`

**Additional context**

I am babysitting a pet piggie!

**Comment by ThilinaRajapakse** [comment]
The combination of a large dataset + a large number of workers can cause issues during tokenization (multiprocessing isn't always stable, sadly). Can you try with `use_multiprocessing=False`?

**Comment by rjurney** [comment]
@ThilinaRajapakse does this use huggingface tokenizers?

**Comment by tshrjn** [comment]
I'm facing a similar issue when training `T5Model`, After around 3 epochs (max_epochs set to 20), the output decoding hangs (not using multiprocessing) & there's still ~100GB of RAM left unused. The eval & output decoding had happened before in previous epochs.

Any clue as to what could have been causing the issue?

**Comment by ThilinaRajapakse** [comment]
That's strange. Can you try without using wandb?
